src/Couchbase.Lite.Tests.Shared/Util/WaitAssert.cs

```diff
@@ -31,8 +32,9 @@ public sealed class WaitAssert
 
         public static void WaitFor(TimeSpan timeout, params WaitAssert[] asserts)
         {
-            foreach (var assert in asserts) {
-                assert.WaitForResult(timeout);
+            var handles = asserts.Select(x => x._mre).ToArray();
+            if (!WaitHandle.WaitAll(handles, timeout)) {
```

Comment:
While this smells bad, I should think that it would actually work... that it would not hang.  I'd be interested to know the conditions under which it did.